supervisor/supervisor.go

```diff
@@ -291,6 +292,14 @@ func (s *Supervisor) Machine() Machine {
 
 // SendTask sends the provided event the the supervisors main event loop
 func (s *Supervisor) SendTask(evt Task) {
+	ctxVal := fmt.Sprintf("%v", evt.Ctx())
+	if evt.Ctx().Done() == nil && ctxVal == "context.Background" {
+		logrus.Debug("Background task found without Done Channel, close.")
+		evt.ErrorCh() <- evt.Ctx().Err()
```

Comment:
is `evt.Ctx().Err()` populated if we are here?

Reply:
No, this particular test never caused this Err to be triggered.  I think that's the root fo the bug, but I don't know how to handle the case where a user does something that won't complete.

I'm at the end of the line for bright ideas on this, if you have any I'd really love to hear them.

Reply:
Oops, my bad typing.  The `evt.Ctx().Err()` field always was nil from what I could see.

Reply:
The code that didn't seem to get triggered that I was thinking about is line 304 `case <-evt.Ctx().Done()`.

I ran this at one point with containerd from /projectatomic/containerd:master rather than projectatomic/containerd:docker-1.13.1-rhel, and I didn't run into any errors, it just seemed to work.  But I'm guessing that will cause other fixed errors to break if we were to use that.